lib/options_page.dart

```diff
@@ -1,37 +1,64 @@
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
 
-List<String> options = [
-  "View Profile",
-  "Message Requests",
-  "Account Settings",
-  "Logout"
-];
-List<Icon> icons = [
-  Icon(Icons.account_circle, size: 45),
-  Icon(Icons.textsms, size: 45),
-  Icon(Icons.settings, size: 45),
-  Icon(Icons.exit_to_app, size: 45)
-];
-
-class optionsPage extends StatefulWidget {
-  @override
-  _optionsPageState createState() => _optionsPageState();
-}
-
-class _optionsPageState extends State<optionsPage> {
+class OptionsPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      body: Center(
-        child: ListView.builder(
-            itemCount: options.length,
-            itemBuilder: (BuildContext context, int index) {
-              return new ListTile(
-                leading: icons[index],
-                title: Text(options[(index)]),
-              );
-            }),
-      ),
-    );
+        appBar: AppBar(),
+        body: ListView(
+          padding: const EdgeInsets.symmetric(vertical: 20),
+          children: <Widget>[
+            ListTile(
+              leading: const Icon(Icons.account_circle, size: 32),
+              title: const Text("View Profile"),
+              onTap: () {
+                debugPrint("Not yet implemented");
+              },
+            ),
+            ListTile(
+              leading: const Icon(Icons.textsms, size: 32),
+              title: const Text("Message Requests"),
+              onTap: () {
+                debugPrint("Not yet implemented");
+              },
+            ),
+            ListTile(
+              leading: Icon(Icons.exit_to_app,
+                  size: 32, color: Theme.of(context).errorColor),
+              title: Text("Logout",
+                  style: TextStyle(color: Theme.of(context).errorColor)),
+              onTap: () {
+                showDialog(
+                    context: context,
+                    builder: (BuildContext dialogContext) {
+                      return AlertDialog(
+                        title: const Text("Logout"),
+                        content: const Text("Are you sure you want to logout?"),
+                        actions: <Widget>[
+                          FlatButton(
+                              child: const Text("CANCEL"),
+                              onPressed: () {
+                                Navigator.pop(dialogContext);
+                              }),
+                          FlatButton(
+                            child: Text("LOGOUT",
+                                style: TextStyle(
+                                    color: Theme.of(context).errorColor)),
+                            onPressed: () async {
+                              await FirebaseAuth.instance.signOut();
```

Comment:
I've always had the idea that we should be signing the user out of the auth provider they signed in with, to avoid the problem that when signing back in, the auth provider sign in step goes straight through and the user can't switch accounts if they want to.  I tested your branch on my iPhone and there was no such problem though (ie. the auth provider sign in asks for the username and password again).  

Here's a SO answer that backs up what I'm saying: https://stackoverflow.com/a/45084649/1992736 

Maybe it has to do with how the auth provider SDK is configured.  

Are there reasons to not sign the user out of the auth provider?   

I'd be more than happy to approve the review and merge this PR and then add an issue to discuss a general solution for the sign out.  There are some further questions like: should we allow users to be signed in to more than one auth provider, can we call sign out on all auth providers even if not signed in or do we need to have access to signed in state... 

Anyway sorry for the brain dump - does that sound OK -> merge this and add a new issue for the auth provider sign out strategy?

Reply:
> I've always had the idea that we should be signing the user out of the auth provider they signed in with, to avoid the problem that when signing back in, the auth provider sign in step goes straight through and the user can't switch accounts if they want to. I tested your branch on my iPhone and there was no such problem though (ie. the auth provider sign in asks for the username and password again).
Here's a SO answer that backs up what I'm saying: https://stackoverflow.com/a/45084649/1992736

You're right about that. I don't know why on iOS it works as intended, but on Android we should call GoogleSignIn.signOut() to present the user with the authentication dialog at the next login attempt.

> Anyway sorry for the brain dump - does that sound OK -> merge this and add a new issue for the auth provider sign out strategy?

Definitely. This kind of work should be part of a separate branch. For now, we just need a quick way to navigate the various pages of the app.